load_tests/load_tests/runner.py

```diff
@@ -0,0 +1,55 @@
+import time
+from typing import Any, List
+
+from locust.events import request_success
+from rasa_integration_testing.interaction import Interaction, InteractionLoader
+from rasa_integration_testing.runner import ScenarioRunner
+from rasa_integration_testing.scenario import Scenario, ScenarioFragmentLoader
+from socketio import Client
+
+from load_tests.constants import SESSION_ID_KEY
+
+EVENT_BOT_UTTERED = "bot_uttered"
+EVENT_USER_UTTERED = "user_uttered"
+
+
+class LocustRunner(ScenarioRunner):
+    def __init__(
+        self,
+        interaction_loader: InteractionLoader,
+        scenario_fragment_loader: ScenarioFragmentLoader,
+    ):
+        self._interaction_loader = interaction_loader
+        self._scenario_fragment_loader = scenario_fragment_loader
+
+    def run(self, scenario: Scenario, client: Client, session_id: str) -> None:
+        interactions: List[Interaction] = self._resolve_interactions(scenario)
+
+        start_time = time.time()
+
+        @client.event
+        def bot_uttered(data):
```

Comment:
is it possible that we receive multiple `bot_uttered` for a single `user_uttered` ?

Reply:
While technically socketio doesn't prevent this, I've never seen Rasa send more than one response per user_utterance. It's probably safe to assume for now that we get bot utterance per user utterance.

Reply:
As discussed, this is the case on websockets. Messages are split into multiple responses